libraries/chain/fork_database.cpp

```diff
@@ -277,20 +277,22 @@ namespace eosio::chain {
          EOS_RETHROW_EXCEPTIONS( fork_database_exception, "serialized fork database is incompatible with configured protocol features" )
       }
 
+      auto prev_head = head_impl(include_root_t::no);
```

Comment:
We don't need this call to `head_impl()` (which may be costly) if `ignore_duplicate == ignore_duplicate_t::no`